public/controller/data_functions.php

```diff
@@ -1058,7 +1061,12 @@ function change_assignment_version($username, $semester,$course, $assignment_id,
         return;
     }
     $json = json_decode(removeTrailingCommas(file_get_contents($file)), true);
-    $json["active_assignment"] = $assignment_version;
+    $json["active_assignment"] = (int)$assignment_version;
+
+    date_default_timezone_set('America/New_York');
+    $version_change_time = date("Y-m-d H:i:s");
+    $version_change = array("version"=>(int)$assignment_version, "time"=>$version_change_time);
+    array_push($json["history"],$version_change);
```

Comment:
The cast was there because when the student press the buttons on the webpage to reset the active version or cancel their submission, the $assignment_version will be a string, whereas uploading a new version will store an int in the $assignment_version. This happened because of the use of htmlspecialchar function in get_assignment_version function.